src/eval_framework/file_utils/file_ops.py

```diff
@@ -0,0 +1,173 @@
+        self._cleanup_temp_dir()
+        # we need to re-raise the signal to terminate gracefully
+        signal.signal(signum, signal.SIG_DFL)
+        os.kill(os.getpid(), signum)
+
+    def _setup_s3_client(self) -> None:
+        required_env_vars = ["AWS_ENDPOINT_URL", "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]
+        for var in required_env_vars:
+            if var not in os.environ:
+                raise ValueError(f"Missing required environment variable: {var}")
+        endpoint = os.environ["AWS_ENDPOINT_URL"]
+        if not endpoint.startswith(("http://", "https://")):
+            os.environ["AWS_ENDPOINT_URL"] = f"https://{endpoint}"
+
+    @property
+    def entity(self) -> str | None:
+        return self.api.default_entity
+
+    def get_artifact(self, artifact_id: str, version: str = "latest") -> wandb.Artifact:
+        return self.api.artifact(f"wandb-registry-model/{artifact_id}:{version}")
+
+    def get_bucket_prefix(self, artifact: str) -> tuple[str, str]:
+        _, bucket, prefix, *_ = urllib.parse.urlparse(artifact)
+        return bucket, prefix
+
+    def ls(self, artifact: wandb.Artifact) -> list[str]:
+        """
+        list all files in the artifact
+
+        Args:
+            artifact: The wandb.Artifact to list files from.
+        Returns:
+            list | The list of file paths in the artifact.
+        """
+        file_list = list(map(lambda x: x.path_uri, [x for x in artifact.files()]))
+        return file_list
+
+    def download_and_use_artifact(
+        self,
+        artifact: wandb.Artifact,
+    ) -> Path:
+        """
+        use_artifact determines the filesystem that the artifact is on, and does one of two things:
+        1. if the file system is remote (s3 type storage) then it will download the artifact to a temp diectory
+           we then call wandb.use_artifact on the artifact that we downloaded.
+        2. if this download fails, we patch boto3 to retrieve the artifact.
+        """
+        # create tempdir
+        artifact_subdir = "/".join(artifact.name.split(":"))
+        if self.user_supplied_download_path is None:
+            temp_dir = tempfile.TemporaryDirectory()
+            self.download_path = Path(temp_dir.name) / artifact_subdir
+            self._temp_dir = temp_dir  # Keep reference to prevent pre-mature cleanup
+        else:
+            assert isinstance(self.user_supplied_download_path, Path)
+            self.download_path = self.user_supplied_download_path / artifact_subdir
+            if self.download_path.exists():
+                wandb.use_artifact(artifact)
+                return self.download_path
+
+        try:
+            artifact_path = artifact.download(root=str(self.download_path))
+            wandb.use_artifact(artifact)
+        except Exception as e:
+            # patch the wandb boto3 call to disable ssl verification
+            try:
+                print(f"failed to download artifact {e}, patching boto3")
```

Comment:
I would remove this message: this will appear all the time and users will not be able to do anything about it. They'll just be triggered and be wondering.